astromon/utils.py

```diff
@@ -310,3 +314,224 @@ def _logging_call_decorator(*args, **kwargs):
 """
 Decorator to add logging messages at the start/end of the decorated function.
 """
+
+
+def calalign_from_files(calalign_dir=None):
+    """
+    Get data from calalign files in `calalign_dir`.
+
+    Parameters
+    ----------
+    calalign_dir: :any:`pathlib.Path`
+        Directory where to find the calalign files.
+
+    Returns
+    -------
+    :any:`astropy.table.Table`
+    """
+    import re
+
+    from astropy.io import fits
+    from astropy.table import Table
+    from cxotime import CxoTime
+
+    if calalign_dir is None:
+        calalign_dir = "/data/caldb/data/chandra/pcad/align"
+
+    calalign_files = sorted(Path(calalign_dir).glob("*.fits"))
+    if not calalign_files:
+        raise Exception(f"CALALIGN files does not exist at {calalign_dir}")
+
+    caldb_info = {
+        "N0008": {
+            "CalDB": "4.4.4",
+            "since": "2011-06-28T20:45:00",
+        },
+        "N0009": {
+            "CalDB": "4.6.2",
+            "since": "2014-07-09T21:00:00",
+        },
+        "N0010": {
+            "CalDB": "4.10.0",
+            "since": "2022-06-28T14:00:00",
+        },
+    }
+
+    transforms = []
+    for filename in sorted(calalign_files):
+        # pcadD2001-01-01alignN0008.fits
+        file_re = re.search(
+            "(?P<date>[0-9]{4}-[0-9]{2}-[0-9]{2})align(?P<version>N[0-9]{4}).fits",
+            filename.name,
+        )
+        if not file_re:
+            raise Exception(f"filename {filename} does not conform to expected format")
+
+        info = file_re.groupdict()
+        hdus = fits.open(filename)
+        cals = hdus[1].data
+        cvsd = (
+            hdus[1].header["CVSD0001"]
+            if "CVSD0001" in hdus[1].header
+            else "2050:001:00:00:00.00"
+        )
+        cved = (
+            hdus[1].header["CVED0001"]
+            if "CVED0001" in hdus[1].header
+            else "2050:001:00:00:00.00"
+        )
+        for cal in cals:
+            transforms.append(
+                {
+                    "start": CxoTime(cvsd),
+                    "stop": CxoTime(cved),
```

Comment:
Nevermind.  I think I see how this works to find the reference caldb.  I suppose for our current align CALDB using CVED is a no-op and it might come up in the future.